Merge from OptimalBits/null

UbiquitousBear opened the pull request.

aleccool213 committed:
1070 Add discard config for rate-limiter

- Meant for Manually Fetching Jobs pattern

GabrielCastro committed:
Catch sub process crashes

ronnievdv committed:
Allow cancelling of retries when using custom backoff strategy

aleccool213 committed:
review comments

alolis committed:
Fix Warning: cancellation is disabled

Bluebird configuration must have cancellation enabled (false by default) in order to be able to use promise.cancel().

sokki committed:
remove the correct listener in isRedisReady

manast committed:
Merge pull request #1096 from wishabi/1070

1070 Add discard config for rate-limiter

manast committed:
Merge pull request #1103 from sokki/patch-1

remove the correct listener in isRedisReady

manast committed:
Merge pull request #1102 from alolis/enable-bluebird-cancelation

Fix Warning: cancellation is disabled

alolis committed:
Reset 'failedReason', 'finishedOn' and 'processedOn' fields on job retry

The class members will be reset to null and removed from Redis database.

ronnievdv committed:
Remove isReady check from test

manast committed:
Merge pull request #1100 from sortdinc/abort_backoff

Allow cancelling of retries when using custom backoff strategy

manast committed:
Merge pull request #1106 from alolis/reset-fields-on-job-retry

Reset 'failedReason', 'finishedOn' and 'processedOn' fields on job retry

manast committed:
Update LICENSE.md

manast committed:
Merge pull request #1091 from GabrielCastro/master

Catch processor crashes

zhaow-de committed:
Make the cron jobs test sequence independent

manast committed:
Merge pull request #1109 from zhaow-de/fix-test_repeat

Make the cron jobs test sequence independent

marcbachmann committed:
Fix typo in forcedReconnection variable

manast committed:
Merge pull request #1114 from marcbachmann/patch-2

Fix typo in forcedReconnection variable

manast committed:
replace bluebird by native promises where possible

manast committed:
remove .only

manast committed:
test: corrected some tests by using bluebird promises

manast committed:
Update issue_template.md

manast committed:
Update issue_template.md

manast committed:
Update issue_template.md

manast committed:
Update issue_template.md

manast committed:
test: skip test that fails only in travis

manast committed:
Merge branch 'master' into native-promises

manast committed:
perf(moveToActive): used local var for rate limiter

manast committed:
merged master

manast committed:
fix(lua): correct numJobs fetch in moveToActive

manast committed:
fix(queue): use Bluebird instances for config and OperationalError

manast committed:
chore: comment un-used parameter

manast committed:
test: used Bluebird instead of Promiser where applicable

manast committed:
test: call done with err where needed

manast committed:
fix(merge): conflict in test

manast committed:
build(travis): added nodejs 10

manast committed:
chore(test): remove console logs

manast committed:
chore(test): removed unused vars

manast committed:
style(lua): minor formatting

manast committed:
fix(test): removed unused undeclared variable

manast committed:
Merge pull request #1119 from OptimalBits/native-promises

replace bluebird by native promises where possible

manast committed:
merged master

manast committed:
fix(delayed): pause delayed jobs #1087

manast committed:
Merge pull request #1125 from OptimalBits/pause-delayed-jobs

Pause delayed jobs, fixes #1087

manast committed:
updated CHANGELOG

manast committed:
build(packages): upgraded several packages

manast committed:
3.5.0

manast committed:
3.5.0

manast committed:
updated CHANGELOG

manast committed:
3.5.1

Commit:
process.NODE_ENV -> process.env.NODE_ENV

Commit:
Disable bluebird cancellation

manast committed:
Merge pull request #1127 from bappogroup/develop

process.NODE_ENV -> process.env.NODE_ENV

manast committed:
- chore(queue): remove bluebird config from the codebase.

manast committed:
doc(changelog): update changelog

manast committed:
test(repeat): skip test that only fails on travis

manast committed:
Merge pull request #1131 from OptimalBits/remove-bluebird-config

Remove bluebird config

manast committed:
Merge pull request #1132 from OptimalBits/develop

release 3.5.2

manast committed:
3.5.2

cemremengu committed:
fix(queue): convert delay setting to number

Fixes #1118

btd committed:
Remove bluebird (#1156)

* Remove bluebird from lib code

* Remove bluebird at all

* Refresh package-lock.json and remove yarn.lock

Commit:
Fix takeLock method - wrong use of "lockKey", using only method lockKey, not property
Extracted checking if queue is closing
Extracted process.nextTick handling

Vortec4800 committed:
Fix breakpoints in child processes

manast committed:
Update README.md

btd committed:
Upgrade codebase with safe es6 features

btd committed:
Merge branch 'develop' of github.com:OptimalBits/bull

manast committed:
fix(doc): fix #1178

manast committed:
Merge pull request #1180 from btd/develop

Upgrade codebase to improve readability

manast committed:
fix: fix #1044 support for typescript processors

manast committed:
Merge pull request #1186 from OptimalBits/fix/fix-typescript-processors

fix: fix #1044 support for typescript processors

manast committed:
Merge branch 'develop' into extracted_methods

manast committed:
chore: fix some lint errors

manast committed:
Merge branch 'rysi3k-extracted_methods' into develop

manast committed:
chore: update ioredis

manast committed:
chore: update CHANGELOG

manast committed:
3.5.3

GabrielCastro committed:
Make progress functions consistently return a promise

manast committed:
fix: properly remove sandbox events

manast committed:
Merge pull request #1190 from OptimalBits/fix/properly-remove-sandbox-events

fix: properly remove sandbox events

manast committed:
Merge pull request #1188 from GabrielCastro/fix/make-progress-consitent

fix: make progress functions in sandbox consistently return a promise

manast committed:
feat: add function to remove repeatable jobs by key

manast committed:
Merge branch 'develop' of github.com:OptimalBits/bull into develop

manast committed:
chore: update CHANGELOG

manast committed:
chore: upgrade dependencies

manast committed:
3.6.0

Commit:
added parameter to optional fetching next job

manast committed:
Update README.md

dkruchinin reviewed (approved):
:shipit: (but let's test it in staging before rolling out)

Commit:
consider priority when promoting a job #1205

Commit:
fix merge hiccup

Commit:
wait for start of job processing to prevent race condition

manast committed:
perf: improve update delay set logic

manast committed:
fix: better handling of update delay set

manast committed:
fix: return a higher resolution timestamp

manast committed:
chore: refactoring updateDelayTimer

manast committed:
test: fixes in test_repeat suite

demidovsky committed:
Skip 'unknown command client' error from worker.js

These errors are thrown when using Google Cloud Platform Redis Memorystore.
As docs say, 'Some commands can interfere with a managed Redis service and are blocked.'
https://cloud.google.com/memorystore/docs/reference/redis-configs#blocked
CLIENT command is in the list.

demidovsky committed:
Merge branch 'develop' of github.com:OptimalBits/bull into google_memorystore_compatibility

manast committed:
Merge pull request #1210 from demidovsky/google_memorystore_compatibility

feat: improve compatibility with Google Cloud Platform Redis Memorystore

manast committed:
Merge pull request #1206 from sokki/fix/promote-with-prio

consider priority when promoting a job #1205

manast committed:
Merge pull request #1207 from OptimalBits/perf/improve-update-delay-set

perf: improve update delay set logic

manast committed:
fix: better delay for rate limited jobs

manast committed:
Merge pull request #1212 from OptimalBits/fix/better-rate-limiting

fix: better delay for rate limited jobs

Vortec4800 committed:
Updated port selection mechanism for child node process inspect flag

Vortec4800 committed:
Merge branch 'develop' into breakpoint

Vortec4800 committed:
Lint fixes after rebase

Vortec4800 committed:
Minor update to updated interpolation syntax

manast committed:
Merge pull request #1168 from Vortec4800/breakpoint

Fix breakpoints in child processes

manast committed:
fix: divide timestamp by 4096 due to changes in lua script

manast committed:
chore: upgrade dependencies

manast committed:
docs: update changelog

manast committed:
3.7.0

0xflotus committed:
fixed performance

manast committed:
Merge pull request #1223 from 0xflotus/patch-1

fixed performance

abriosi committed:
Edit `guardInterval` description

When running multiple concurrent workers with delayed tasks, the default value of `guardInterval` will cause spikes on network bandwidth, cpu usage and memory usage. Each concurrent worker will run the delayed job watchdog. In case of multiple workers set this value to something much higher, e.g. `guardInterval = numberOfWorkers*5000`

manast committed:
Merge pull request #1224 from abriosi/patch-1

Edit `guardInterval` description

foxling committed:
Ignore unknown command client error.
On some cloud services, such as Alibaba Cloud or Tencent Cloud,
the error message is "ERR unknown command ' client '".

stefanpl committed:
Correct call signature for 'process' in guide

manast committed:
Merge pull request #1242 from stefanpl/improved-guide-docs

Correct call signature for 'process' in guide

manast committed:
Update README.md

Eomm committed:
fix: typo PATTERNS.md

manast committed:
Merge pull request #1249 from Eomm/develop

fix: typo PATTERNS.md

GabrielCastro committed:
Update sandbox exit handler to log signals

manast committed:
Merge pull request #1252 from GabrielCastro/fix/better-exit-logging

Update sandbox exit handler to log signals

Eomm committed:
add: lockRenewTime in AdvancedSettings

gabegorelick committed:
Don't use `latest` for get-port version

This fixes churn in package-lock.json as well as preventing
future breakage when get-port releases a backwards incompatible
version.

The `get-port` dependency was added in #1168.

manast committed:
Merge pull request #1259 from gabegorelick/get-port-version

Don't use `latest` for get-port version

manast committed:
Merge pull request #1256 from Eomm/develop

docs: lockRenewTime in AdvancedSettings

minzojian committed:
Update worker.js

sometimes, the 'ERR unknown command \'client\'' check is not enough, there is another like "ERR unknown command ' client '". there are spaces around the "client" word.

ArnaudValensi committed:
Improve Job#progress reference

zhaow-de committed:
fixed the wrong invocation parameters in a dead function - Scripts#moveToFailed

manast committed:
Merge pull request #1265 from ArnaudValensi/patch-1

Improve Job#progress reference

GabrielCastro committed:
docs: add link to prometheus exporter

zhaow-de committed:
moved scripts#cleanJobsInSet to Lua

zhaow-de committed:
handled the promise rejection to capture an exception in test_rate_limiter.js

zhaow-de committed:
removed a redundant parameter at Job#addJob

manast committed:
Merge pull request #1271 from zhaow-de/redundant-parameter-at-addjob

removed a redundant parameter at Job#addJob

manast committed:
Merge pull request #1269 from zhaow-de/move-to-lua-script

moved scripts#cleanJobsInSet to Lua

manast committed:
Merge pull request #1268 from GabrielCastro/docs/link-to-exporter

docs: add link to prometheus exporter

manast committed:
Merge pull request #1263 from minzojian/develop

Update worker.js with another check for ERR known command

foxling committed:
Merge branch 'develop' into unknown_command_client_compatibility

manast committed:
Merge pull request #1240 from foxling/unknown_command_client_compatibility

Ignore unknown command client error.

manast committed:
fix: remove broken child processes fix #1098

manast committed:
Merge pull request #1191 from rysi3k/optional_fetching_next_job

Added parameter to optional fetching next job

manast committed:
doc: updated moveToComplete params

manast committed:
Merge pull request #1290 from OptimalBits/fix/remove-broken-child-process-fix1098

fix: remove broken child processes fix #1098

manast committed:
fix: return every in getRepeatableJobs #1284

manast committed:
doc: add removeRepeatableByKey doc

manast committed:
Merge pull request #1291 from OptimalBits/fix/fix-1284

fix: return every in getRepeatableJobs #1284

Commit:
Storing finished date in job object

Commit:
Clean up delay timer

manast committed:
Merge pull request #1292 from rysi3k/store_finished_date

Storing finished date in job object

manast committed:
Merge pull request #1294 from peakon/fix/delay_timer

Rewrite timing logic for updating delay set

MarkHerhold committed:
Grammar Fixes

manast committed:
feat: add commitlint

manast committed:
docs: updated readme

manast committed:
docs: update changelog

manast committed:
3.8.0

manast committed:
fix: wait for ready in cleanJobsInSet fixes #1298

manast committed:
docs: update changelog

manast committed:
3.8.1

manast committed:
Merge pull request #1297 from MarkHerhold/patch-1

README Grammar Fixes

manast committed:
feat: add job.log #1165

manast committed:
Merge pull request #1307 from OptimalBits/feat/add-logging-for-jobs-1165

feat: add job.log #1165

dextermb committed:
Fix typo in Queue#clean example

- `job -> jobs`

manast committed:
Merge pull request #1309 from dextermb/patch-1

Fix typo in Queue#clean example

manast committed:
docs: update CHANGELOG

manast committed:
ci: upgrade dependencies

manast committed:
3.9.0

manast committed:
fix: add job log method to job wrapper

manast committed:
docs: update changelog

manast committed:
ci: bump to version 3.9.1

manast committed:
fix: add more files and dirs to npmignore

ianp committed:
Use correct type for !== comparison

ianp committed:
fix: include dot in file extensions

Because this is how `path.extname()` returns them.

manast committed:
Merge pull request #1320 from ianp/issue-1251

Use correct type for !== comparison

manast committed:
ci: drop support for node 6

manast committed:
feat: support for keeping specified number of completed/failed jobs

manast committed:
Merge pull request #1331 from OptimalBits/feat/add-keep-complete-fai-jobs-261

Feat/add keep complete fai jobs 261

manast committed:
fix: remove logs when autoremoving jobs fixes #1330

manast committed:
Merge pull request #1332 from OptimalBits/fix/delete-job-logs-with-autoremove

fix: remove logs when autoremoving jobs fixes #1330

manast committed:
docs: update changelog

manast committed:
ci: upgrade get-port and p-timeout

manast committed:
3.10.0

stansv committed:
docs: fixed global listener example

manast committed:
Merge pull request #1333 from stansv/develop

docs: fixed global listener example

darky committed:
add bull-real to UIs

gabegorelick committed:
fix: whenCurrentJobsFinished shouldn't initialize bclient

Fixes #1346

gabegorelick committed:
test: fix race condition in test_connection.js

gabegorelick committed:
fix: unhandled promise warning

By returning null, a warning won't be generated about unhandled
promise.

alexandrugheorghe committed:
Update REFERENCE.md

Removed copied arguments section from Job#log

manast committed:
Merge pull request #1370 from alexandrugheorghe/patch-1

Fixed REFERENCE.md faulty Job#log description

rosslavery committed:
Update clientCommandMessageReg to allow for command enclosed in backticks

manast committed:
Merge pull request #1376 from rosslavery/fix/issue-1375

Update clientCommandMessageReg to allow for command enclosed in backticks

manast committed:
Merge pull request #1359 from gabegorelick/unhandled-promise-warning

fix: unhandled promise warning

manast committed:
Merge pull request #1347 from gabegorelick/pause-bclient

fix: whenCurrentJobsFinished shouldn't initialize bclient

manast committed:
Merge pull request #1337 from darky/develop

add bull-repl to UIs

Commit:
When job data are updated, save it also in current job object

Commit:
Merge branch 'develop' of github.com:OptimalBits/bull into store_updated_data_in_job

alexandrugheorghe committed:
Added caution note to repeatable jobs

Commit:
feat: basic support for bulk adding jobs

manast committed:
Merge pull request #1419 from peakon/feature/bulk_add

Support for bulk adding jobs

manast committed:
Merge pull request #1267 from zhaow-de/wrong-paramater-in-a-dead-function

fixed the wrong invocation parameters in a dead function

manast committed:
Merge pull request #1402 from alexandrugheorghe/improvement/readme

Added caution note to repeatable jobs section

manast committed:
Merge pull request #1382 from rysi3k/store_updated_data_in_job

When job data are updated, save it also in current job object

manast committed:
Merge pull request #1270 from zhaow-de/unhandled-promise-rejection-in-test

handled the promise rejection to capture an exception in test_rate_limiter.js

SSARCandy committed:
fix typo

manast committed:
Merge pull request #1448 from SSARCandy/patch-1

fix typo in document

collinwu committed:
fix sandboxed processor header link in readme

stansv committed:
Merge pull request #1449 from collinwu/fix-doc-link-1

fix sandboxed processor header link in readme

Commit:
Adds bull-board to list of UIs

manast committed:
Merge pull request #1451 from joaomilho/patch-1

Adds bull-board to list of UIs

pandres95 committed:
Fix-[Sandbox] If the child process is killed, remove it from the pool.

manast committed:
Merge pull request #1461 from Valiuapp/develop

Fix-[Sandbox] If the child process is killed, remove it from the pool.

darky committed:
typo fix

manast committed:
Merge pull request #1469 from darky/patch-1

typo fix

adisreyaj committed:
Updated the Job#moveToFailed type

Updated the Job#moveToFailed function parameter's data types.

manast committed:
Merge pull request #1471 from adisreyaj/patch-1

Updated the Job#moveToFailed type

manast committed:
chore: cleaner code for isFinished

manast committed:
test: minor formatting

manast committed:
Merge branch 'develop' of github.com:OptimalBits/bull into develop

manast committed:
chore: update CHANGELOG

manast committed:
ci: updated dependencies

manast committed:
3.11.0

manast committed:
Update issue_template.md

Commit:
feat: support async custom backoffs

akirilyuk committed:
revert yarn.lock

manast committed:
Merge pull request #1490 from akirilyuk/feature/should-be-able-to-handle-custom-async-backoff

feat: support async custom backoffs

manast committed:
Update README.md

manast committed:
Update README.md

manast committed:
Merge pull request #1514 from OptimalBits/develop

Merge develop into master

PengXing committed:
docs: update REFERENCE.md (#1844)

animir committed:
docs: add a list of useful job properties to REFERENCE (#1797)

kevgathuku committed:
docs(reference): add missing `createClient` option in the `QueueOptions` reference

manast committed:
feat(types): add typescript types to package

Barrierml committed:
docs: fix the missing brackets

DavideViolante committed:
docs: Update PATTERNS.md (#2721)

redis config needs 2 params since bull v4
